### PR TITLE
Add support for wrong fixture path outside a bundle

### PR DESCRIPTION
**Pull request by Kwadz**
Related to #301

### CODE REVIEW

File: Test/WebTestCase.php [hunk 1]
```diff
@@ -520,6 +522,9 @@ private function locateResources($paths)
                 $files[] = $path;
                 continue;
             }
+            if ($path[0] !== '@' && !file_exists($path) === true) {
```
Comment (alexislefebvre) [inline Test/WebTestCase.php]: Is there a reason you didn't put this in an `else` block?

**Comment by Kwadz** [comment]
I wanted to avoid nested `if` statements and keep parameters symmetric. But I agree with your code, in this case it doesn't make more sense having every parameters written 2 times.

**Comment by Kwadz** [comment]
So, we go for the following?
```php
if ($path[0] !== '@') {
    if (!file_exists($path)) {
        throw new \InvalidArgumentException(sprintf('Unable to find file "%s".', $path));
    }
    $files[] = $path;
    continue;
}
```

**Comment by alexislefebvre** [comment]
Sorry for thinking to this so late but, could you please add a corresponding test? Otherwise, I'll add a test by myself, that's not a problem. Thanks.

**Comment by alexislefebvre** [comment]
Thanks for the test! I'm sorry to bother you, but can you please add a test that load successfully a fixture without the `@` at the beginning ?

I'm thinking to something like this:

```php
    /**
     * Use nelmio/alice.
     */
    public function testLoadFixturesFilesWithPathOutsideABundle()
    {
        $fixtures = $this->loadFixtureFiles(array(
            '[…]/Tests/App/DataFixtures/ORM/user.yml',
        ));
        $this->assertInternalType(
            'array',
            $fixtures
        );
        // 10 users are loaded
        $this->assertCount(
            10,
            $fixtures
        );
    }
```

Feel free to to add a fixture in another directory if this is required by the test.

See this test for reference: https://github.com/liip/LiipFunctionalTestBundle/blob/97cedfc1edd51e142044f2db3759bd96ef7a9d5a/Tests/Test/WebTestCaseTest.php#L365-L425 (I don't think that we perform all the tests, counting the number of loaded fixtures is sufficient)

**Comment by Kwadz** [comment]
The `testLoadFixturesFilesPaths()` method already do the job. The returned value by Symfony Kernel is an absolute path.

https://github.com/liip/LiipFunctionalTestBundle/blob/master/Tests/Test/WebTestCaseTest.php#L422

**Comment by alexislefebvre** [comment]
Thanks for your answer.

Yes, `testLoadFixturesFilesPaths()` will load the fixtures, but the test uses `$this->client->getContainer()->get('kernel')->locateResource(` with a `@` in the path, so I think that it is different from your original problem.

I added a test that doesn't rely on `locateResource()` and a `@…` fixture.

**Comment by Kwadz** [comment]
In fact `$this->client->getContainer()->get('kernel')->locateResource()` is only used for test purpose. It returns the desired absolute path which will be used by `WebTestCase::loadFixtureFiles()` and therefore by `WebTestCase::locateResources()`. The goal is not to test Symfony Kernel in our case but `WebTestCase::locateResources()`. I think there is no need for additional test.

**Comment by alexislefebvre** [comment]
Yes, this test was not necessary, I just wanted to ensure that the behaviour described in your issue #301 worked.

**Pull request merged**